src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Abstractions/IMacroDependency.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Abstractions
+{
+    /// <summary>
+    /// Represents a dependency configuration for a macro.
+    /// </summary>
+    public interface IMacroConfigDependency
+    {
+        /// <summary>
+        /// Gets the set of dependencies required by the macro.
```

Comment:
```suggestion
        /// Gets the set of symbol names required by the macro.
```

Reply:
`ResolveSymbolDependencies` method should be called prior to accessing the property. Before it the property won't be populated.

---

It would be good to add exception if `Dependencies` is accessed prior to `ResolveSymbolDependencies` method call.

Reply:
done